src/lavinmq/mqtt/session.cr

```diff
@@ -43,8 +43,10 @@ module LavinMQ
           Fiber.yield if (i &+= 1) % 32768 == 0
         rescue ::IO::Error
           @log.error { "deliver loop exited due to IO error" }
+          break
         rescue ArgumentError
           @log.error { "deliver loop exited due to argument error" }
```

Comment:
I think the exception message should be included in the error. Maybe even log the entire error?